### PR TITLE
add: function that exposes profile names

### PR DESCRIPTION
**Pull request by kiiadi**
## Motivation and Context

Tooling will often want the ability to list available profiles for someone to select from. Currently the `ProfileSet` does not expose this.

## Description

Added `fn profiles()` function to `ProfileSet`

## Testing

New unit test added

## Checklist

- [ ] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/rust-runtime/aws-config/src/profile/parser.rs [hunk 1]
```diff
@@ -132,6 +132,11 @@ impl ProfileSet {
         self.profiles.is_empty()
     }
 
+    /// Returns the names of the profiles in this profile set
+    pub fn profiles(&self) -> impl Iterator<Item=&str> {
+        self.profiles.keys().into_iter().map(|k| k.as_ref())
+    }
```
Comment (Velfi) [inline aws/rust-runtime/aws-config/src/profile/parser.rs:138]:
```suggestion
    pub fn profiles(&self) -> impl Iterator<Item=&str> {
        self.profiles.keys().map(String::as_ref)
    }
```

File: aws/rust-runtime/aws-config/src/profile/parser.rs [hunk 2]
```diff
@@ -235,21 +240,29 @@ mod test {
...
+    #[test]
+    fn profile_names_are_exposed() {
+        let source = make_source(ParserInput {
+            config_file: Some("[profile foo]\n[profile bar]".to_string()),
+            credentials_file: Some("".to_string())
+        });
+
+        let profile_set = ProfileSet::parse(source).expect("profiles loaded");
+
+        let mut profile_names: Vec<&str> = profile_set.profiles().collect();
```
Comment (Velfi) [inline aws/rust-runtime/aws-config/src/profile/parser.rs:261]:
```suggestion
        let mut profile_names: Vec<_> = profile_set.profiles().collect();
```

You only need to specify the container when collecting (Vec is container in this case)

**Comment by Velfi** [comment]
Make sure you have `precommit` installed per the main README.md
it'll run the formatter for you, plus some other stuff

File: aws/rust-runtime/aws-config/src/profile/parser.rs (hunk 2 above)
Comment (kiiadi) [reply aws/rust-runtime/aws-config/src/profile/parser.rs:261]: OK cool - will fix.

**Review by Velfi — APPROVED** [review]

**Pull request merged**